waterlog/src/Components/Map.js

```diff
@@ -1,42 +1,44 @@
-import React, { Component } from 'react';
-import {
-  Map,
-  TileLayer,
-  Polyline,
-  CircleMarker,
-  Popup
-} from 'react-leaflet';
-import PropTypes from 'prop-types';
-import { connect } from 'react-redux';
-import { fetchMapsData } from './../actions/MapActions';
-import Loader from './Loader';
+import React, { Component } from "react";
+import { Map, TileLayer, Polyline, CircleMarker, Popup } from "react-leaflet";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { fetchMapsData } from "./../actions/MapActions";
+import Loader from "./Loader";
+import Error404 from "./Error404";
 
-const backgroundColor = '#253238';
-const errorColor = '#FF1744';
-const lighterColor = '#4F5B62';
+const backgroundColor = "#253238";
+const errorColor = "#FF1744";
+const lighterColor = "#4F5B62";
 
 function formatMapData(data) {
   let markers, segments;
   let todaysLeaks;
   if (!data.leaks) {
     todaysLeaks = [];
   } else {
-    todaysLeaks = data.leaks.filter(leak => (
-      (new Date(leak.latestTimeStamp)).getDay() === (new Date(Date.now())).getDay() &&
-      (new Date(leak.latestTimeStamp)).getMonth() === (new Date(Date.now())).getMonth() &&
-      (new Date(leak.latestTimeStamp)).getFullYear() === (new Date(Date.now())).getFullYear()
-    ));
+    todaysLeaks = data.leaks.filter(
+      leak =>
+        new Date(leak.latestTimeStamp).getDay() ===
```

Comment:
those are a lot of repeated calls look at putting `Date(Date.now())` into a variable

Reply:
done